src/encoder.rs

```diff
@@ -0,0 +1,5 @@
+use crate::droplet::Droplet;
+
+pub trait Encoder {
```

Comment:
What's the reasoning for adding this trait. I see it being implemented, but unless I'm missing something, I don't see it being used anywhere.

Reply:
Both `RobustEncoder` and `IdealEncoder` use it. I changed them from implementing an `Iterator` to this, questionable at best but atleast they now return a `Droplet` everytime instead of an `Option<Droplet>` as they used to.